py2exe/runtime.py

```diff
@@ -307,11 +306,20 @@ def build(self):
     def get_runstub_bytes(self, target):
         from sysconfig import get_platform
         if target.exe_type in ("console_exe", "service"):
-            run_stub = 'run%s-py%s.%s-%s.exe' % (RUNTIME_SUFFIX, sys.version_info[0], sys.version_info[1], get_platform())
+            if 'GCC' in sys.version:
+                run_stub = 'run%s-py%s%s-%s.exe' % (RUNTIME_SUFFIX, sys.version_info[0], sys.version_info[1], get_platform())
+            else:
+                run_stub = 'run%s-py%s.%s-%s.exe' % (RUNTIME_SUFFIX, sys.version_info[0], sys.version_info[1], get_platform())
         elif target.exe_type == "windows_exe":
-            run_stub = 'run_w%s-py%s.%s-%s.exe' % (RUNTIME_SUFFIX, sys.version_info[0], sys.version_info[1], get_platform())
+            if 'GCC' in sys.version:
+                run_stub = 'run_w%s-py%s%s-%s.exe' % (RUNTIME_SUFFIX, sys.version_info[0], sys.version_info[1], get_platform())
+            else:
+                run_stub = 'run_w%s-py%s.%s-%s.exe' % (RUNTIME_SUFFIX, sys.version_info[0], sys.version_info[1], get_platform())
         elif target.exe_type == "ctypes_comdll":
-            run_stub = 'run_ctypes_dll%s-py%s.%s-%s.dll' % (RUNTIME_SUFFIX, sys.version_info[0], sys.version_info[1], get_platform())
+            if 'GCC' in sys.version:
+                run_stub = 'run_ctypes_dll%s-py%s%s-%s.dll' % (RUNTIME_SUFFIX, sys.version_info[0], sys.version_info[1], get_platform())
+            else:
+                run_stub = 'run_ctypes_dll%s-py%s.%s-%s.dll' % (RUNTIME_SUFFIX, sys.version_info[0], sys.version_info[1], get_platform())

```

Comment:
This code is already quite ugly. With this (although necessary for you) change, it becomes almost unreadable. Since you only need to remove an extra `'.'`, I would suggest to define a string like
```
cpython_version_separator = '.' if 'GCC' not in sys.version else ''
```
and then put it in each of the the format strings, e.g.
```
run_stub = 'run_w%s-py%s%s%s-%s.exe' % (RUNTIME_SUFFIX, sys.version_info[0], cpython_version_separator, sys.version_info[1], get_platform())
```

Reply:
(if you prefer, you can also refactor the whole thing using f-strings, since now we support only 3.8+)